src/address.rs

```diff
@@ -1,8 +1,16 @@
+use halo2::pasta::pallas;
+
 use crate::keys::Diversifier;
 
 /// A shielded payment address.
 #[derive(Debug)]
 pub struct Address {
     d: Diversifier,
-    pk_d: (),
+    pk_d: pallas::Point,
+}
+
+impl Address {
+    pub(crate) fn from_parts(d: Diversifier, pk_d: pallas::Point) -> Self {
+        Address { d, pk_d }
+    }
 }
```

Comment:
As mentioned in `keys.rs`, instead of a new method `from_parts` this could be accomplished with `impl From<(Diversifier, DiversifiedTransmissionKey)> for Address`

Reply:
This is an internal API, for which I prefer to be explicit.

Reply:
For Zebra usage this may not stay internal

Reply:
If it does become necessary to expose it for Zebra, we can do so in a subsequent PR.